prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/git/GitClone.java

```diff
@@ -0,0 +1,85 @@
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+
+import static com.redhat.parodos.workflow.context.WorkContextDelegate.getOptionalValueFromRequestParams;
+import static com.redhat.parodos.workflow.context.WorkContextDelegate.getRequiredValueFromRequestParams;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import java.io.File;
+import java.util.List;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.InvalidRemoteException;
+import org.eclipse.jgit.api.errors.TransportException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@AllArgsConstructor
+public class GitClone extends BaseWorkFlowTask {
+
+	@Autowired
+	private GitUtils gitUtils;
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key(gitUtils.getUri()).type(WorkParameterType.TEXT).optional(false)
+						.description("Url to clone from").build(),
+				WorkParameter.builder().key(gitUtils.getBranch()).type(WorkParameterType.TEXT).optional(true)
+						.description("Branch to clone from, default main").build(),
+				WorkParameter.builder().key("credentials").type(WorkParameterType.TEXT).optional(false)
+						.description("Git credential").build());
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String gitUri = "";
+		String destination = "";
+		String gitBranch = "";
+
+		try {
+			gitUri = getRequiredValueFromRequestParams(workContext, gitUtils.getUri());
+			gitBranch = getOptionalValueFromRequestParams(workContext, gitUtils.getBranch(), "main");
+			destination = cloneRepo(gitUri, gitBranch);
+		}
+		catch (MissingParameterException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+		catch (TransportException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new Exception("cannot connect to the repository server"));
+		}
+		catch (InvalidRemoteException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new Exception("Remote repository " + gitUri + " is not available"));
+		}
+		catch (GitAPIException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("cannot clone repository"));
+		}
+
+		workContext.put(gitUtils.getContextUri(), gitUri);
+		workContext.put(gitUtils.getContextDestination(), destination);
+		workContext.put(gitUtils.getContextBranch(), gitBranch);
+		return new DefaultWorkReport(WorkStatus.COMPLETED, workContext, null);
+	}
+
+	private String cloneRepo(String gitUri, String gitBranch)
+			throws InvalidRemoteException, TransportException, GitAPIException {
+		String tmpdir = Files.createTempDir().getAbsolutePath();
+		Git.cloneRepository().setURI(gitUri).setBranch("refs/heads/" + gitBranch).setDirectory(new File(tmpdir)).call();
```

Comment:
Why use ` Files.createTempDir().getAbsolutePath()` and then `new File(tmpdir)` ?
Since `Files.createTempDir()` returns `File`, is it possible to use only that?